service_container/factories.rst

```diff
@@ -293,5 +293,149 @@ previous examples takes the ``templating`` service as an argument:
                 ->args([ref('templating')])
             ;
         };
+        
+Usage example
+-------------
+
+The following example is intended to show how to create and use a factory method in Symfony framework.
+Suppose you want to realize the factory method pattern for services, that describe two delivery methods - DHL and UPS.
+
+Services (subclasses) definition::
+
+    // src/Deliveries/DeliveryInterface.php
+    namespace App\Deliveries;
+	
+    interface DeliveryInterface
+    {
+    }
+
+    // src/Deliveries/DHL.php
+    namespace App\Deliveries;
+	
+    use App\Deliveries\DeliveryInterface;
+
+    class DHL implements DeliveryInterface
+    {
+        public $costLabel;
+    
+        public function cost() {
+            return 100;
+        }
+    }
+    
+    // src/Deliveries/UPS.php
+    namespace App\Deliveries;
+	
+    use App\Deliveries\DeliveryInterface;
+
+    class UPS implements DeliveryInterface
+    {
+        public $costLabel;
+    
+        public function cost() {
+            return 200;
+        }
+    }
+
+Factory definition::
+
+    // src/Factories/DeliveryFactory.php
+    namespace App\Factories;
+	
+    use App\Deliveries\DeliveryInterface;
+    
+    abstract class DeliveryFactory
```

Comment:
An abstract class cannot be used as a factory.

Reply:
I'm afraid, I don't understand you... Matt Zandstra gives factory method's example with abstract class in his book. And example below works in Symfony without any exceptions. May I ask for the details of your comment?

Reply:
It currently works because the factory method happens to be static, but that example code is very fragile.

Reply:
Changed it in my today pull request.